### PR TITLE
Don't leak file descriptor in error cases

### PR DESCRIPTION
**Pull request by praiskup**
This is wasting fd numbers if compress operates recursively.

### CODE REVIEW

File: compress.c [hunk 1]
```diff
@@ -597,30 +596,27 @@ void comprexx(char *fileptr)
...
 						perror(ofname);
-						exit_code = 1;
-						return;
+						goto error;
 					}
 				}
 
 				if ((fdout = open(ofname, O_WRONLY|O_CREAT|O_EXCL|O_BINARY, 0600)) == -1) {
 					perror(tempname);
-					return;
```
Comment (praiskup) [inline compress.c]: Note that here was not set `exit_code = 1`, while it IMO should be.

Comment (vapier) [reply compress.c]: that sounds fine

File: compress.c [hunk 2]
```diff
@@ -751,6 +746,14 @@ void comprexx(char *fileptr)
 				tempname);
 			break;
 	}
+
+	return;
+
+error:
+	exit_code = 1;
+	close (fdin);
```
Comment (vapier) [inline compress.c]: no space before the (

File: compress.c [hunk 3]
```diff
@@ -751,6 +746,14 @@ void comprexx(char *fileptr)
...
+	if (fdout != -1)
+		close (fdout);
```
Comment (vapier) [inline compress.c]: same here

**Comment by vapier** [comment]
oh this is the master branch.  don't worry about it for now.

**Comment by praiskup** [comment]
Can I reopen then (with fixes)?  I'd be really afraid of regressions otherwise.

**Pull request closed without merging**